src/app/pipes/meql.pipe.spec.ts

```diff
@@ -0,0 +1,356 @@
+  }
+
+  public static doubleConditionTest(pipe: MeqlPipe, query: any) {
+    /*
+  Expected example:
+
+  (
+    "Field" = "Value"
+    and "Field2" = "Value2"
+  )
+   */
+    const line1 = `(${newline}`;
+    const line2 = `${tab}"${query.rules[0].field}" ${query.rules[0].operator} "${query.rules[0].value}"${newline}`;
+    const line3 = `${tab}${query.condition} "${query.rules[1].field}" ${query.rules[1].operator} "${query.rules[1].value}"${newline}`;
+    const line4 = ')';
+    return [pipe.transform(query), `${line1}${line2}${line3}${line4}`];
+  }
+
+  public static doubleRulesetTest(pipe: MeqlPipe, query: any) {
+    /*
+  Expected example:
+
+  (
+    "Field" = "Value"
+    and "Field2" = "Value2"
+    and (
+        "Field3" = "Value3"
+        or "Field4" = "Value4"
+    )
+  )
+   */
+    const line1 = `(${newline}`;
+    const line2 = `${tab}"${query.rules[0].field}" ${query.rules[0].operator} "${query.rules[0].value}"${newline}`;
+    const line3 = `${tab}${query.condition} "${query.rules[1].field}" ${query.rules[1].operator} "${query.rules[1].value}"${newline}`;
+    const line4 = `${tab}${query.condition} (${newline}`;
+    const line5 = `${tab}${tab}"${query.rules[2].rules[0].field}" ${query.rules[2].rules[0].operator} "${query.rules[2].rules[0].value}"${newline}`;
+    const line6 = `${tab}${tab}${query.rules[2].condition} "${query.rules[2].rules[1].field}" ${query.rules[2].rules[1].operator} "${query.rules[2].rules[1].value}"${newline}`;
+    const line7 = `${tab})${newline}`;
+    const line8 = ')';
+    return [
+      pipe.transform(query),
+      `${line1}${line2}${line3}${line4}${line5}${line6}${line7}${line8}`,
+    ];
+  }
+}
+
+describe('JqlPipe', () => {
+  let pipe: MeqlPipe;
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      providers: [MeqlPipe],
+    });
+    pipe = TestBed.inject(MeqlPipe);
+  });
+
+  it('create an instance', () => {
+    expect(pipe).toBeTruthy();
+  });
+
+  it('transforms field equal to string', () => {
+    const testResult = meqlTestHelper.singleConditionTest(
+      pipe,
+      'Simple String Field',
+      '=',
+      'Some Text'
+    );
+    expect(testResult[0]).toBe(testResult[1]);
+  });
+
+  it('transforms field not equal to string', () => {
+    const testResult = meqlTestHelper.singleConditionTest(
+      pipe,
+      'Simple String Field',
+      '!=',
+      'Some Text'
+    );
+    expect(testResult[0]).toBe(testResult[1]);
+  });
+
+  it('transforms field equal to number', () => {
+    const testResult = meqlTestHelper.singleConditionTest(
+      pipe,
+      'Simple String Field',
+      '=',
+      0
+    );
+    expect(testResult[0]).toBe(testResult[1]);
+  });
+
+  it('transforms field not equal to number', () => {
+    const testResult = meqlTestHelper.singleConditionTest(
+      pipe,
+      'Simple String Field',
+      '!=',
+      99
+    );
+    expect(testResult[0]).toBe(testResult[1]);
+  });
+
+  it('transforms and uses "and" condition', () => {
```

Comment:
Another Jest feature is you can nest `describe()` calls. Here I see you have lots of tests but maybe they could be grouped together:

```ts
describe('MeqlPipe', () => {
  describe('single condition tests', () => {
    // it() calls go here...
  });

  describe('AND condition tests', () => {
    // it() calls go here...
  });

  // etc...
});
```

Not necessary, depends on how you want to structure your tests.

Reply:
Test groups created. It looks neater and is easier to follow.